### DISCUSSION TITLE
Add type_trait for IDL types, so as to get PubSubType from idl types.

### DISCUSSION THREAD

**Discussion by wjswjq** [eProsima/Fast-DDS#3394] [Ideas]
Now it is easy to get idl type from PubSubType by calling PubSubType::type.  I suggest to add some method like type_trait so that we can get PubSubType from idl types. This might simplify the usage as the user does not need to know about PubSubType any more. 

- common header

```
template <typename MessageT>
struct type_traits {
  static ::eprosima::fastdds::dds::TopicDataType* GetPubSubType() {
    return nullptr;
  }
};
```

-  codes generated by fastddsgen
```
template <>
struct type_traits<test> {
  static ::eprosima::fastdds::dds::TopicDataType* GetPubSubType() {
    return new testPubsubTypes();
  }
};
```

**Comment by Mario-DL** [comment]
Hi @wjswjq 

Thanks for the suggestion ! While it could be a good idea to isolate the `<idltype>PubSubType()` from the user, I can come with some debatable points:
- Method's name: since `Get..()` (for me) its declaring the intention of retrieving some internal object, not creating a new one every time. 
- Given the case, it could be arguable if constructor for the `PubSubType()` should be hidden so that creation is managed from only one place, which I don`t know if it would be standard-compliant right now.

**Reply by wjswjq** [reply]
- the method name could be discussed.
- Later maybe the TypeSupport class can be constructed by idl type， rather than PubSubType？

**Reply by Mario-DL** [reply]
I don't think that the `TypeSupport` should be constructed directly from the `idl_type` but from a  `TopicDataType`  since (among other things) the implementation for methods like `serialize()` or `deserialize()` for the eProsima's types (`SerializedPayload_t`) must be provided.
The idea is that the user-defined types should be kept agnostic of the vendor-specific types which is the reasoning for using an intermediate type to accommodate both (`...PusSubTypes()`))

**Reply by wjswjq** [reply]
And maybe method name could be something like `NewPubSubType()` ?
- common header
```
template <typename MessageT>
struct type_traits {
  static ::eprosima::fastdds::dds::TopicDataType* NewPubSubType() {
    return nullptr;
  }
};
```
- codes generated by fastddsgen
```
template <>
struct type_traits<test> {
  static ::eprosima::fastdds::dds::TopicDataType* NewPubSubType() {
    return new testPubsubTypes();
  }
};
```
